spec/factories/user.rb

```diff
@@ -2,8 +2,8 @@
 
 FactoryBot.define do
   factory :user do
-    sequence(:uid) { FFaker::Internet.user_name }
-    sequence(:email) { FFaker::Internet.email }
+    sequence(:uid) { FFaker::InternetSE.login_user_name }
```

Comment:
I switched to this as it removes the period from the username, which is breaking the `user_url` method.  This alternative method in the FFaker gem accounts for it - https://github.com/ffaker/ffaker/blob/main/lib/ffaker/internet_se.rb#L33

Reply:
wow! what a catch!